go/libkb/api.go

```diff
@@ -319,22 +320,32 @@ func (a *InternalAPIEngine) isExternal() bool { return false }
 var lastUpgradeWarningMu sync.Mutex
 var lastUpgradeWarning *time.Time
 
-func (a *InternalAPIEngine) consumeHeaders(resp *http.Response) error {
+func (a *InternalAPIEngine) consumeHeaders(resp *http.Response) (err error) {
 	u := resp.Header.Get("X-Keybase-Client-Upgrade-To")
 	p := resp.Header.Get("X-Keybase-Upgrade-URI")
-	if len(u) > 0 {
-		a.G().NotifyRouter.HandleClientOutOfDate(u, p)
+	m := resp.Header.Get("X-Keybase-Upgrade-Message")
+	if len(u) > 0 || len(m) > 0 {
+		a.G().NotifyRouter.HandleClientOutOfDate(u, p, m)
```

Comment:
This was part of my confusion. `keybase ping` doesn't seem to be a "real" API call, so this never happens in that case, though the client parses the header (I think). But in the case of logout/login, this path is executed many times, and this NotifyRouter call is not throttled by the 3 minute window, below. If that's the intent, it all seems to work OK.